gulpfile.js

```diff
@@ -1,60 +1,104 @@
 const gulp = require("gulp");
+const del = require("del");
 const sass = require("sass");
 const fs = require("fs");
+const path = require("path");
 const rename = require("gulp-rename");
 const transform = require("gulp-transform");
 const buildContent = require("./src/content");
 const Viz = require('viz.js');
 const vizRenderOpts = require('viz.js/full.render.js');
 
+const runServer = require("./server");
+
+const paths = {
+  dist: "./dist",
+  distAssets: "./dist/assets/",
+  srcPages: "./src/content/**/readme.md",
+  srcResources: "./src/content/**/*.@(jpg|jpeg|png|gif)",
+  srcDiagrams: "./src/content/**/*.@(dot|neato|fdp|sfdp|twopi|circo)",
+  srcContentBase: "./src/content",
+  srcAssetImages: "./src/@(assets)/**/*.@(jpg|jpeg|png|gif)",
+  srcStyleEntry: "./src/assets/style.scss",
+  srcStylesAny: "./src/assets/**/*.scss",
+  vendorAssets: ["./node_modules/highlight.js/styles/atom-one-dark.css"],
+  invaderTagDefsBase: "./lib/invader/src/tag/hek/definition/",
+};
+
+//the dist directory may contain outdated content, so start clean
+function clean() {
+  return del(paths.dist);
+}
```

Comment:
It's important that this is run as a part of the default build so that before publishing to S3 we can avoid including any files which were previously built but then refactored to new paths.